### DISCUSSION TITLE
forge build not compiling

### DISCUSSION THREAD

**Discussion by mihirkansal** [Cyfrin/foundry-full-course-cu#2351] [Q&A] [answered]
Hi! running forge build is returning "Nothing to compile" Can someone please point out what I'm doing wrong?

![image](https://github.com/user-attachments/assets/719540a1-2875-405b-b215-e0d02dbfce11)

**Comment by EngrPips** [comment, marked as answer]
Hello @gitofrivia, Please run `forge init` on an empty directory to initiate a proper `Foundry` repo.

**Reply by mihirkansal** [reply]
Is this necessary? Because Patrick runs forge build in the course without using forge init

**Reply by EngrPips** [reply]
Maybe he didn't show it, but he must have initiated a proper `Foundry` repo. I have always done it that way and never run into this issue you are running into.
